src/resolve/resolved-dns-query.c

```diff
@@ -1140,3 +1140,10 @@ bool dns_query_fully_confidential(DnsQuery *q) {
 
         return FLAGS_SET(q->answer_query_flags, SD_RESOLVED_CONFIDENTIAL) && !q->previous_redirect_non_confidential;
 }
+
+bool dns_query_fully_synthetic(DnsQuery *q) {
+        assert(q);
+
+        return FLAGS_SET(q->answer_query_flags, SD_RESOLVED_SYNTHETIC) &&
+                (q->answer_query_flags & SD_RESOLVED_FROM_MASK) == 0;
```

Comment:
This would also exclude SD_RESOLVED_FROM_TRUST_ANCHOR, which I think should have the AA bit, same as anything we synthesize.

So maybe use a test like `(q->answer_query_flags & (SD_RESOLVED_FROM_TRUST_ANCHOR|SD_RESOLVED_SYNTHETIC)) && !(q->answer_query_flags & SD_RESOLVED_FROM_CACHE|SD_RESOLVED_FROM_ZONE|SD_RESOLVED_FROM_NETWORK)` ?